src/main/java/de/numcodex/sq2cql/model/Mapping.java

```diff
@@ -98,4 +110,12 @@ public Map<TermCode, AttributeMapping> attributeMappings() {
     public Optional<String> timeRestrictionPath() {
         return Optional.ofNullable(timeRestrictionPath);
     }
+
+    public Optional<String> codeFhirPath() {
+      return Optional.ofNullable(codeFhirPath);
+    }
+
+    public Optional<TermCode> primaryCode() {
```

Comment:
Please explain what the `primaryCode` is including an example.

Reply:
We also could return the `key` here if we have no `primaryCode`, because the key should be always or old unmapped term code - right?

Reply:
The primaryCode is an addition to work with resources whose termCode cannot be applied to the primaryCodePath (https://raw.githubusercontent.com/cqframework/clinical_quality_language/master/Src/java/quick/src/main/resources/org/hl7/fhir/fhir-modelinfo-4.0.1.xml) as is done by a retrieve statement. Instead of the termCode the primaryCode is used. In prior versions, the key fortunately (by chance) always was applicable to the primaryCodePath.
The here presented solution is an intermediate solution. Moving forward each map entry should have a primaryCode (which admittedly will in many cases be the key) but if no primaryCode is available we should not default back to the key but use a retrieve on the resource. Moving forward we should only evaluate the termCode if a codeFhirPath is present.
This gives the maximum degree of freedom.